compiler/test/stdlib/number.test.gr

```diff
@@ -93,6 +93,143 @@ assert Number.isNaN(Number.div(Number.nan, Number.nan))
+assert Number.pow(-0.0, 0.5) == 0.0
+assert Number.isNaN(Number.pow(-0.0, 0.0))
+assert Number.isNaN(Number.pow(-0.0, -0.0))
+assert Number.pow(-0.0, -0.5) == Number.infinity
+assert Number.pow(-0.0, -1.0) == Number.infinity * -1
+assert Number.pow(-0.0, -2.0) == Number.infinity
+assert Number.pow(-0.0, -3.0) == Number.infinity * -1
+assert Number.pow(-0.0, -4.0) == Number.infinity
+assert Number.pow(-0.0, Number.infinity * -1) == Number.infinity
+assert Number.isNaN(Number.pow(Number.nan, 0.0))
+assert Number.isNaN(Number.pow(Number.infinity, 0.0))
+assert Number.isNaN(Number.pow(Number.infinity * -1, 0.0))
+assert Number.isNaN(Number.pow(1.0, 0.0))
+assert Number.isNaN(Number.pow(-1.0, 0.0))
+assert Number.isNaN(Number.pow(-0.5, 0.0))
+assert Number.isNaN(Number.pow(Number.nan, -0.0))
+assert Number.pow(1, 1) == 1
+assert Number.pow(2, 1) == 2
+assert Number.pow(300, 1) == 300
+assert Number.pow(-1, 1) == -1
+assert Number.pow(-2, 1) == -2
+assert Number.pow(-300, 1) == -300
+assert Number.pow(0, 1) == 0
+assert Number.pow(1, 0) == 1
+assert Number.pow(0, 0) == 1
+assert Number.pow(1, 5) == 1
+assert Number.pow(5, 5) == 3125
+assert Number.pow(-5, 5) == -3125
+assert Number.pow(5, 6) == 15625
+assert Number.pow(-5, 6) == 15625
+assert Number.pow(1.0, 1) == 1.0
+assert Number.pow(2, 1.0) == 2.0
+assert Number.pow(300.0, 1) == 300.0
+assert Number.pow(-1, 1.0) == -1.0
+assert Number.pow(-2.0, 1) == -2.0
+assert Number.pow(-300, 1.0) == -300.0
+assert Number.pow(0.0, 1) == 0.0
+assert Number.isNaN(Number.pow(1, 0.0))
+assert Number.isNaN(Number.pow(0.0, 0))
+assert Number.pow(1, 5.0) == 1.0
+assert Number.pow(5.0, 5) == 3125.0
+assert Number.pow(-1.0, 1) == -1.0
+assert Number.pow(-4.0, 3) == -64.0
+assert Number.pow(-4, 3.0) == -64.0
+assert Number.pow(-5, 5.0) == -3125.0
+assert Number.pow(5.0, 6) == 15625.0
+assert Number.pow(-5, 6.0) == 15625.0
+assert Number.pow(1/2, 1) == 1/2
+assert Number.pow(1/2, 2) == 1/4
+assert Number.pow(2, 1/2) == 1.4142135623730952
+assert Number.pow(4, 1/2) == 2
+assert Number.pow(4, -1/2) == 0.5
+assert Number.pow(9223372036854775809, 2) ==
+85070591730234615884290395931651604481
+assert Number.pow(9223372036854775809, 0.5) == 3037000499.97605
+assert Number.pow(9223372036854775809.0, 0.5) == 3037000499.97605
+assert Number.pow(9223372036854775809, 0) == 1
+// TODO(#1168): Refactor This once rationals are fixed
+let denom = 85070591730234615884290395931651604481
+assert Number.pow(9223372036854775809, -2) ==
+1 / denom // 1.1754943508222875e-38
+assert Number.pow(9223372036854775809, 1/5) == 6208.375056426594
+assert Number.pow(9223372036854775809, 2/3) == 4398046511103.9927
+assert Number.pow(10223372036854775809, 10) ==
+12472159440978016923768615307032788210916694000775261660538874886865415760948494778813645195039710006678013364969179502650466497057008288260604039903029954443675868581729857084924132550246401
+// Exp - These may differ from other languages as other languages use an approximation
```

Comment:
Don't think we need the approximation note
```suggestion
// exp
```

Reply:
Alright blaine was just asking why they differed from js

Reply:
I think it was just a PR question :)